### DISCUSSION TITLE
i2C send multiple bytes

### DISCUSSION THREAD

**Discussion by TangYanYee** [felias-fogg/SoftI2CMaster#82] [Q&A] [answered]
i want to do sth similar to Wire.write(data, length) in wire.h. How should I do this? Should I do sth same as the following code?
for(int i = 0; i<length;i++){
  i2c_start(address);
  i2c_write(x[i]);
  i2c_stop();
}
This seems only the first byte is received from my other board.

**Comment by felias-fogg** [comment, marked as answer]
Hi,
the loop for writing the bytes should be between the start and stop call.

**Reply by TangYanYee** [reply]
Ok thankyou for your help!
